eva_submission/nextflow/accession_and_load.nf

```diff
@@ -323,37 +324,103 @@ process load_vcf {
-        pipeline_parameters += " --app.vep.version=" + vep_version.toString()
-        pipeline_parameters += " --app.vep.path=" + "${params.vep_path}/ensembl-vep-release-${vep_version}/vep"
-        pipeline_parameters += " --app.vep.cache.version=" + vep_cache_version.toString()
-        pipeline_parameters += " --app.vep.cache.species=" + vep_species.toString()
+    """
+    java -Xmx4G -jar $params.jar.eva_pipeline --spring.config.location=file:$params.load_job_props --parameters.path=$params.load_job_props $pipeline_parameters
+    """
+}
+
+
+/*
+ * Load into variant db.
+ */
+process run_vep_on_variants {
+    clusterOptions {
+        return "-o $params.logs_dir/annotation.${vcf_filename}.log \
+                -e $params.logs_dir/annotation.${vcf_filename}.err"
     }
 
+    input:
+    tuple val(vcf_filename), val(vcf_file), val(fasta), val(analysis_accession), val(db_name), val(vep_version), val(vep_cache_version), val(vep_species), val(aggregation)
+    val variant_load_complete
+
+    output:
+    val true, emit: vep_run_complete
+
+    memory '5 GB'
+
+    script:
+    def pipeline_parameters = ""
+
+    pipeline_parameters += " --spring.batch.job.names=annotate-variants-job"
+
+    pipeline_parameters += " --input.vcf.aggregation=" + aggregation.toString().toUpperCase()
+    pipeline_parameters += " --input.vcf=" + vcf_file.toRealPath().toString()
+    pipeline_parameters += " --input.vcf.id=" + analysis_accession.toString()
+    pipeline_parameters += " --input.fasta=" + fasta.toString()
+
+    pipeline_parameters += " --spring.data.mongodb.database=" + db_name.toString()
+
+    pipeline_parameters += " --annotation.skip=false"
+    pipeline_parameters += " --app.vep.version=" + vep_version.toString()
+    pipeline_parameters += " --app.vep.path=" + "${params.vep_path}/ensembl-vep-release-${vep_version}/vep"
+    pipeline_parameters += " --app.vep.cache.version=" + vep_cache_version.toString()
+    pipeline_parameters += " --app.vep.cache.species=" + vep_species.toString()
+
     """
     java -Xmx4G -jar $params.jar.eva_pipeline --spring.config.location=file:$params.load_job_props --parameters.path=$params.load_job_props $pipeline_parameters
     """
 }
 
 
+
+/*
+ * Load into variant db.
+ */
+process calculate_statistics_vcf {
+    clusterOptions {
+        return "-o $params.logs_dir/statistics.${vcf_filename}.log \
+                -e $params.logs_dir/statistics.${vcf_filename}.err"
+    }
+
+    input:
+    tuple val(vcf_filename), val(vcf_file), val(fasta), val(analysis_accession), val(db_name), val(vep_version), val(vep_cache_version), val(vep_species), val(aggregation)
+    val variant_load_complete
+
+    output:
+    val true, emit: statistics_calc_complete
+
+    memory '5 GB'
+
+    script:
+    def pipeline_parameters = ""
+
+
+    pipeline_parameters += " --spring.batch.job.names=calculate-statistics-job"
+
+    pipeline_parameters += " --input.vcf.aggregation=" + aggregation.toString().toUpperCase()
```

Comment:
If I'm understanding the [old job configurations](https://github.com/tcezard/eva-pipeline/blob/EVA3455_variantload/src/main/java/uk/ac/ebi/eva/pipeline/configuration/jobs/AggregatedVcfJobConfiguration.java) correctly, we should only run statistics calculation for genotyped vcf.

Reply:
That's a good point and one I was not aware of.
I'm a bit surprised by this considering that there are both variant level and study level statistics in the mongodb database. Maybe the study level is calculated in the Variant Load step.

Reply:
I wonder if this is why [this assertion](https://github.com/EBIvariation/eva-pipeline/pull/194/files/bf0662ac1f1d79a0b57fe418cde3edac5371747b..b4478b79f6dd865b8847c870ea02c976083dec89#diff-bba4f9ffe691665f28c563045033dedeecf355ebb7b424ed3b9b3d05b1dd6951R133) passes in your test, even though the stats step hasn't been run...

Reply:
Yes that might be something else to investigate